ClientGUI/XNAChatTextBox.cs

```diff
@@ -35,10 +32,10 @@ protected override bool HandleKeyPress(Keys key)
             {
                 if (currentNode == null)
                 {
-                    if (enteredMessages.First == null)
-                        return true;
-
-                    currentNode = enteredMessages.First;
+                    if (enteredMessages.First != null)
+                    {
+                        currentNode = enteredMessages.First;
+                    }
```

Comment:
Not sure why those are braced.
```suggestion
                        currentNode = enteredMessages.First;
```